cli.js

```diff
@@ -26,23 +27,27 @@ if (argv.help) {
 
 var paths = []
 
-if (argv.c) {
-  if (typeof argv.c === 'string') {
-    paths.push(`.env.${argv.c}.local`, `.env.${argv.c}`)
-  }
-  paths.push(".env.local", ".env")
-}
-
 if (argv.e) {
   if (typeof argv.e === 'string') {
     paths.push(argv.e)
   } else {
     paths.push(...argv.e)
   }
+} else {
+  paths.push('.env')
 }
 
-if (paths.length === 0) {
-  paths.push('.env')
+if (argv.c) {
+  paths = paths.reduce((acc, p) => acc.concat(
+    typeof argv.c === 'string'
+      ? [`${p}.${argv.c}.local`, `${p}.${argv.c}`, `${p}.local`, p]
+      : [`${p}.local`, p]
```

Comment:
pitty we cannot use flatMap yet but this works as well. 


1 suggestion though: use longer names as that helps with readability (path over p, accumulator over acc)
```suggestion
  paths = paths.reduce((accumulator, path) => accumulator.concat(
    typeof argv.c === 'string'
      ? [`${path}.${argv.c}.local`, `${path}.${argv.c}`, `${path}.local`, p]
      : [`${path}.local`, path]
```

Reply:
re: flatMap - definitely! That would have been much clearer but I had checked the `engines` field in dotenv’s `package.json` and noted they supported node.js `>=8` so thought I’d best fall back to reduce to not exclude anyone unnecessarily.

Happy with the longer names—adds some clarity.